File: 0274-h-index/0274-h-index.java
class Solution {
    public int hIndex(int[] citations) {
        Arrays.sort(citations);
        reverse(citations);
        for(int i = 0; i < citations.length; i++){
            if(i+1>citations[i]){
                return i;
            }
        }
        return citations.length;
    }
    public static void reverse(int[] arr) {
        int left = 0, right = arr.length - 1;
        while (left < right) {
            int temp = arr[left];
            arr[left] = arr[right];
            arr[right] = temp;
            left++;
            right--;
        }
    }
}

File: 0274-h-index/0274-h-index.cpp
class Solution {
public:
    int hIndex(vector<int>& citations) {
        vector<int> bucket(citations.size() + 1, 0);
        for(int n : citations){
            n < bucket.size() ? bucket[n]++ : bucket[citations.size()]++;
        }

        int countSum = 0;
        for(int i = bucket.size() - 1; i > 0; i--){
            countSum += bucket[i];
            if(countSum >= i){
                return i;
            }
        }
        return citations[0] == 0 ? 0 : 1;
    }
};
